analytics_data_api/management/commands/generate_data.py

```diff
@@ -0,0 +1,339 @@
+
+
+def get_count(start):
+    delta = 25 * random.gauss(0, 1)
+    return int(start + delta)
+
+
+class FakeDataCommand(BaseCommand, metaclass=abc.ABCMeta):
+
+    @abc.abstractmethod
+    def add_arguments(self, parser):
+        pass
+
+    def generate_daily_data(self, course_id, start_date, end_date, database,
+                            delete_data=True, add_birth_year=True, use_current_cumulative=False):
+        # Use the preset ratios below to generate data in the specified demographics
+
+        gender_ratios = {
+            'm': 0.6107,
+            'f': 0.3870,
+            'o': 0.23
+        }
+        education_level_ratios = {
+            'associates': 0.058,
+            'bachelors': 0.3355,
+            'primary': 0.0046,
+            'secondary': 0.2442,
+            'junior_secondary': 0.0286,
+            'masters': 0.2518,
+            'none': 0.0032,
+            'other': 0.0271,
+            'doctorate': 0.0470
+        }
+        country_ratios = {
+            'US': 0.33,
+            'GH': 0.12,
+            'IN': 0.10,
+            'CA': 0.14,
+            'CN': 0.22,
+            'DE': 0.08,
+            'UNKNOWN': 0.01
+        }
+
+        enrollment_mode_ratios = {
+            'audit': 0.15,
+            'credit': 0.15,
+            'honor': 0.25,
+            'professional': 0.10,
+            'verified': 0.35
+        }
+
+        if add_birth_year:
+            # Generate birth year ratios
+            birth_years_range = list(range(1960, 2005))
+            ratios = [n / 1000.0 for n in constrained_sum_sample_pos(len(birth_years_range), 1000)]
+            birth_years = dict(list(zip(birth_years_range, ratios)))
+
+        if delete_data:
+            # Delete existing data
+            for model in [models.CourseEnrollmentDaily,
+                          models.CourseEnrollmentModeDaily,
+                          models.CourseEnrollmentByGender,
+                          models.CourseEnrollmentByEducation,
+                          models.CourseEnrollmentByBirthYear,
+                          models.CourseEnrollmentByCountry,
+                          models.CourseProgramMetadata]:
+                model.objects.using(database).all().delete()
+
+            logger.info("Deleted all daily course enrollment data.")
+            logger.info("Generating new daily course enrollment data...")
+
+        logger.info("Generating new daily course enrollment data...")
+
+        # Create new data
+        daily_total = 1500
+        date = start_date
+
+        enrollment_objs = models.CourseEnrollmentModeDaily.objects.using(database).filter(course_id=course_id)
+        cumulative_count = enrollment_objs.latest().cumulative_count \
+            if (use_current_cumulative and enrollment_objs) \
+            else 0
+
+        progress = tqdm(total=(end_date - date).days + 2)
+        while date <= end_date:
+            daily_total = get_count(daily_total)
```

Comment:
this chunk could be profitably split out to increase readability, it's generating one day